packages/core/src/RenderingEngine/vtkClasses/vtkStreamingOpenGLTexture.js

```diff
@@ -239,6 +176,78 @@ function vtkStreamingOpenGLTexture(publicAPI, model) {
   publicAPI.setUpdatedFrame = (frameIndex) => {
     model.updatedFrames[frameIndex] = true;
   };
+
+  const superModified = publicAPI.modified;
+  publicAPI.modified = () => {
+    // Really important piece here that wasted a lot of time from me
+    // always make sure to call the super method first
+    superModified();
+    const volume = cache.getVolume(model.volumeId);
+    const imageIds = volume.imageIds;
+
+    for (let i = 0; i < imageIds.length; i++) {
+      model.updatedFrames[i] = true;
+    }
+  };
+
+  publicAPI.hasUpdatedFrames = () => {
+    return model.updatedFrames.some((frame) => frame);
+  };
+
+  function updateTextureImagesUsingVoxelManager(updatedFrames) {
```

Comment:
This seems a lot cleaner now.  It becomes fairly obvious how to handle updating other data structures such as too large image sets given some of this behaviour.